octobot_trading/exchanges/connectors/ccxt/ccxt_adapter.py

```diff
@@ -63,11 +63,16 @@ def fix_ohlcv(self, raw, **kwargs):
         if "time_frame" in kwargs:
             candles_s = common_enums.TimeFramesMinutes[common_enums.TimeFrames(kwargs["time_frame"])] * \
                         common_constants.MINUTE_TO_SECONDS
-        for ohlcv in fixed:
+        for index, ohlcv in enumerate(fixed):
             try:
                 int_val = int(self.get_uniformized_timestamp(ohlcv[common_enums.PriceIndexes.IND_PRICE_TIME.value]))
                 ohlcv[common_enums.PriceIndexes.IND_PRICE_TIME.value] = int_val - (int_val % candles_s)
                 self._fix_ohlcv_prices(ohlcv)
+            except TypeError:
+                # the last candle might not be properly set
+                if index == len(fixed) - 1 and self.connector.exchange_manager.exchange.DUMP_INCOMPLETE_LAST_CANDLE:
```

Comment:
Should we reverse the condition to first check if `self.connector.exchange_manager.exchange.DUMP_INCOMPLETE_LAST_CANDLE`?

Reply:
right